backend/src/utils.rs

```diff
@@ -0,0 +1,46 @@
+use subxt::ext::sp_core::{crypto::SecretStringError, Pair};
+
+use crate::{
+    configuration::Configuration,
+    error::AppError,
+    tx::kilt::{self, runtime_types::did::did_details::DidSignature},
+};
+
+pub async fn get_current_block(config: &Configuration) -> Result<u64, subxt::Error> {
+    let api = config.get_client().await?;
+    let block_number = api
+        .rpc()
+        .block(None)
+        .await
+        .map_err(|e| format!("Failed to get block number: {e}"))?
+        .ok_or("Failed to get block number")?
+        .block
+        .header
+        .number;
+    Ok(block_number)
+}
+
+pub async fn get_tx_counter(config: &Configuration) -> Result<u64, AppError> {
```

Comment:
should be called `get_next_tx_counter` because of the +1 it does

Reply:
[d6cdd01](https://github.com/KILTprotocol/attester/pull/1/commits/d6cdd01c07154b8319da4b6a297dde7a297371e4)